test/test_console_tool.py

```diff
@@ -566,6 +566,183 @@ def test_get_bucket_show_size(self):
             '''
             self._run_command(['get-bucket', '--showSize', 'my-bucket'], expected_stdout, '', 0)
 
+    def test_get_bucket_with_versions(self):
+        self._authorize_account()
+        self._create_my_bucket()
+
+        # Put many versions of a file into the test bucket. Unroll the loop here for convenience.
+        bucket = self.b2_api.get_bucket_by_name('my-bucket')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+
+        # Now check the output of get-bucket against the canon.
+        expected_stdout = '''
+        {
+            "accountId": "my-account",
+            "bucketId": "bucket_0",
+            "bucketInfo": {},
+            "bucketName": "my-bucket",
+            "bucketType": "allPublic",
+            "corsRules": [],
+            "fileCount": 10,
+            "lifecycleRules": [],
+            "revision": 1,
+            "totalSize": 40
+        }
+        '''
+        self._run_command(['get-bucket', '--showSize', 'my-bucket'], expected_stdout, '', 0)
+
+    def test_get_bucket_with_folders(self):
+        self._authorize_account()
+        self._create_my_bucket()
+
+        # Create a hierarchical structure within the test bucket. Unroll the loop here for
+        # convenience.
+        bucket = self.b2_api.get_bucket_by_name('my-bucket')
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
```

Comment:
Why so many files?  Not that it's a real problem, but I might have stopped at a depth of 3 or 4.  I think the point of this test is to make sure that it walks files inside folders, and once you're down a couple levels that probably proves the point.

Reply:
It probably doesn't add anything to test 6 levels instead of 3, so the decision is somewhat arbitrary. Structural induction is probably a reasonable argument for stopping ~3-4, but hypothetically a strange implementation might introduce problems at some levels and not others. Problems are probably more likely to occur at enormous depths, if at all. Since all other things are essentially equal, 6 seemed as good as 3, so that's just what I happened to do.